golang/pkg/builder/container/container_builder.go

```diff
@@ -164,7 +164,17 @@ func (dc *DockerContainerBuilder) WithLogConfig(logConfig *container.LogConfig)
 
 // Sets the image of a container in a "image:tag" format where image can be a fully qualified name.
 func (dc *DockerContainerBuilder) WithImage(imageWithTag string) *DockerContainerBuilder {
-	dc.imageWithTag = imageWithTag
+	image, err := imageHelper.URIFromString(imageWithTag)
+	if err != nil {
+		panic(err) // TODO(robot9706): how to handle correctly?
```

Comment:
Risky, there are no recovers in grpc commucation :disappointed: either we add recover functions or change builder in a way, that these check goes into the create fn, that could throw an error.

Reply:
why not just store the error and return it on the build method or something?

Reply:
I'm going to move the parsing into the Create function so we can return the error there.